mne/time_frequency/tfr.py

```diff
@@ -99,6 +99,59 @@ def morlet(sfreq, freqs, n_cycles=7.0, sigma=None, zero_mean=False):
         Ws.append(W)
     return Ws
 
+def wavelet(sfreq, freqs, n_cycles=7.0, window=np.hanning):
+    """Compute wavelets for the given frequency range with the specified windowing function.
+
+        Parameters
+        ----------
+        sfreq : float
+            The sampling Frequency.
+        freqs : array
+            Frequency range of interest (1 x Frequencies).
+        n_cycles : float | array of float, default 7.0
+            Number of cycles. Fixed number or one per frequency.
+        window : Callable object
```

Comment:
I would make `window` anything that is accepted by `get_window`:

https://docs.scipy.org/doc/scipy/reference/generated/scipy.signal.get_window.html

And then below we just call `get_window(window, n)` and we don't have to do any sanity checking (SciPy will do it for us)